src/passa/internals/_pip.py

```diff
@@ -315,3 +322,76 @@ def prepare(self):
 
     def install(self):
         self.wheel.install(self.paths, distlib.scripts.ScriptMaker(None, None))
+
+
+def _iter_egg_info_directories(root, name):
+    name = packaging.utils.canonicalize_name(name)
+    for parent, dirnames, filenames in os.walk(root):
+        matched_indexes = []
+        for i, dirname in enumerate(dirnames):
+            if not dirname.lower().endswith(".egg-info"):
```

Comment:
to be exhaustive you can (and should) actually just check for `.endswith("egg-info")` as sometimes -- e.g. if pip prepares this -- the directory might just be called `pip-egg-info`